pkg/worker/workerhandler/claimresolvehandler/custom_ensure.go

```diff
@@ -17,38 +19,83 @@ import (
 func (h *SystemHandler) Ensure(tas *task.Task, bud *budget.Budget) error {
 	var err error
 
-	//     find claims with lifecycle "propose" that are expired
-	var cla []*poststorage.Object
+	// find trees with lifecycle "propose" that are expired
+	var trees []*poststorage.Object
 	{
-		cla, err = h.sto.Post().SearchExpiry()
+		trees, err = h.sto.Post().SearchExpiry()
 		if err != nil {
 			return tracer.Mask(err)
 		}
 	}
```

Comment:
Let's try to agree on some terminology here. The result of `SearchExpiry` is a list of post objects in apiserver speak. Those post objects represent granular claims. Each of them has a tree ID. Each of them has the lifecycle `propose`. Our goal in this worker handler is to do onchain and offchain management so that we can progress in the lifecycle of the whole tree. The "propose" claims that expired need a new "resolve" claim once the onchain magic is properly configured. So at the end we need to make sure that a new post object with kind `claim` and lifecycle `resolve` is created with the same tree ID as the claim that is processed respectively.

Reply:
https://github.com/uvio-network/apiserver/pull/49/commits/78c529ee124d5aeb25b3528a209cb71ac17e7a54